docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,412 @@
+$ conduit config
+```
+
+</details>
+
+<br/>
+
+<details>
+<summary><code style="font-size: 19px; font-weight:bold;">conduit run [...]</code></summary>
+
+#### Description
+
+- This command will run Conduit with all the configured pipelines, connectors, etc.
+- It is equivalent to the current `conduit` command.
+- `config.path` will be the root of the working enviornment. Example:
+
+```bash
+$ pwd
+/usr/code
+
+$ ls
+conduit.yaml
+connectors/
+pipelines/
+processors/
+```
+
+- Other flags such as `connectors.path`, etc. will overwrite the existing configuration on `conduit.yaml`. This will need to be evaluated before specifying to conduit to accomodate both scenarios (absolute and relative paths). 
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| connectors.path | path to standalone connectors' directory | No | "./connectors" |
+| db.badger.path | path to badger DB | No | "conduit.db" |
+| processors.path | path to standalone processors' directory | No | "./processors" |
+| pipelines.path | path to the directory that has the yaml pipeline configuration files, or a single pipeline configuration file | No | "./pipelines" |
+| db.postgres.connection-string | postgres connection string | Yes |  |
+| db.postgres.table | postgres table in which to store data | No | "conduit_kv_store" |
+| db.type | database type; accepts badger,postgres,inmemory | No | "badger" |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+| http.address | address for serving the HTTP API | No | ":8080" |
+| log.format | sets the format of the logging; accepts json, cli | No | "cli" |
+| log.level | sets logging level; accepts debug, info, warn, error, trace | No | "info" |
+| pipelines.exit-on-error | exit Conduit if a pipeline experiences an error while running | No |  |
+
+#### `--help`
+
+```bash
+$ conduit run
+```
+</details>
+
+<br/>
+
+<details>
+<summary><code style="font-size: 19px; font-weight:bold;">conduit pipelines init [NAME] [--pipelines.path]</code></summary>
+
+#### Description
+
+- This command will initialize a pipeline in the previously configured pipelines path (existing on `conduit.yaml`). In other words, even if you aren't on the pipelines directory, this pipeline will be initialized there.
+- In the event of not having a `conduit.yaml` configuration file already, we should prompt to initialize a working conduit environment.
+- It does not require having conduit running.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| name  |  Pipeline file name and pipeline name  | No | `pipeline-#` (`pipeline-#.yaml`) |
+
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| destination  |  Plugin name of the destination connector  | No | `log` |
+| pipelines.path  |  Where to initialize a new pipeline | No | `.` (current directory) |
```

Comment:
WDYT about path? For two reasons I'd say: (1) because of the other flags (destination and source don't have that prefix), and because of the command itself (pipelines).

Reply:
@hariso my thinking is that we'll have other paths out there and would like to make it explicit. Especially considering there will be a global `config.path`. Want to make sure they're not confused.

Reply:
@hariso I think that if we want to preserve consistency, maybe adding `pipelines.path` to a pipelines command is the way to go considering the global flag for configuration will have to be `--config.path`. I also think that would likely make a better design if all flags were simply `--path`, but I'm leaning towards being explicit without limiting ourselves in the future in the event of having to combine both path flags for some reason.

Reply:
Hopefully, by using configuration files, this is not a flag users need to type much anyways, so I would think it's a decent tradeoff.